source/game/system/Random.hh

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+/// Based off https://github.com/em-eight/mkw/blob/master/source/game/util/Random.cpp
+
+#include <Common.hh>
+
+namespace System {
+
+class Random {
+public:
+    Random(u32 seed) : m_x(seed), m_seed(seed) {}
+
+    ~Random() = default;
+
+    [[nodiscard]] u32 nextInt() {
```

Comment:
We run into a problem with this being nodiscard. Let's create a new `next` function that only bumps `m_x` that returns void.